Sources/LanguageServerProtocolTransport/DisableSigpipe.swift

```diff
@@ -10,35 +10,32 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if canImport(Glibc)
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
 import Glibc
 #elseif canImport(Musl)
 import Musl
 #elseif canImport(Android)
 import Android
 #endif
 
-#if canImport(Glibc) || canImport(Musl) || canImport(Android)
+#if !os(Windows)
 // This is a lazily initialised global variable that when read for the first time, will ignore SIGPIPE.
 private let globallyIgnoredSIGPIPE: Bool = {
-  /* no F_SETNOSIGPIPE on Linux :( */
+  // No F_SETNOSIGPIPE on Linux
```

Comment:
Instead of globally disabling `SIGPIPE` on Darwin, shouldn’t we use `F_SETNOSIGPIPE` to just disable it on the pipes that we care about, ie. calling

```swift
fcntl(/*fileDescriptor*/, F_SETNOSIGPIPE)
```
 
I feel like that would be a cleaner solution unless you have a good reason to disable `SIGPIPE` everywhere.

Not that it has a lot of information but `man fcntl` is the most detailed explanation I found of this.

Reply:
> I feel like that would be a cleaner solution unless you have a good reason to disable SIGPIPE everywhere.

Is there a case where we'd *want* to crash from a SIGPIPE? I could see some argument that we should be resetting SIGPIPE for subprocesses, but given we're a long lived service, we should (IMO) just be handling the inevitable I/O errors everywhere (which I believe we are?). Plus not diverging between macOS and Linux/Windows (Windows doesn't even have SIGPIPE) seems ideal to me anyway.

Reply:
Hmm, that’s a fair point. I just thought that we get more specific error messages on a pipe failure if we set `F_SETNOSIGPIPE` but maybe it’s just easier and more consistent to just ignore SIGPIPE everywhere, you’re right.